### DISCUSSION TITLE
how to format a number in engineering notation

### DISCUSSION THREAD

**Discussion by dominiklenne** [freeplane/freeplane#450] [General]
I would really be interested in how to format a number in engineering notation, i.e. with the exponent as a multiple of 3, using the 

- node properties pane - format - node text    

input field.  I tried

%.1e        - yields scientific notation
%###.1e - yields no exponential notation at all

I looked up:

https://stackoverflow.com/questions/37679136/format-double-in-engineering-notation-with-fixed-number-of-fractional-digits
http://www.java2s.com/Code/Java/Data-Type/DecimalFormatE0exponentmustbemultipleof3.htm

but obviously my understanding of how java number formatting relates to freeplane is not sufficient (i am not a java programmer btw).

thx for any hint... cheers...

**Comment by vboerchers** [comment]
Hi Dominik,

actually I don't unterstand what you mean by "exponent as a multiple of 3", but this is the full spec for the Formatter: http://docs.oracle.com/javase/7/docs/api/java/text/DecimalFormat.html

to test things quickly use a formula like this: `=new java.text.DecimalFormat("%.1e").format(0.12345)`

Regards, Volker

**Reply by dominiklenne** [reply]
thx for the reply. i meant things like
12,300        --> 12.3e3
12,300,000 --> 12.3e6
0.123          --> 123e-3
problem is solved now, see next answer and my answer if interested.

**Comment by EdoFro** [comment]
input this:   `##0.#####E0`  as Core Text Format

it's explained in the page that @vboerchers recommended:
https://docs.oracle.com/javase/7/docs/api/java/text/DecimalFormat.html

and look in the **Scientific Notation** subtitle.

before:
![image](https://user-images.githubusercontent.com/35700575/173394751-19d5a73e-6a15-474c-beae-67ce0aa76a1d.png)

after:
![image](https://user-images.githubusercontent.com/35700575/173394974-1c65db8c-374d-4cb0-9cdc-46d7905317f8.png)

**Reply by dominiklenne** [reply]
thx!
i had read
https://www.freeplane.org/wiki/index.php/Data_recognition_and_data_formats#Number_formats
and there, i concluded, that to get an [exponential representation according to java standards](https://stackoverflow.com/questions/37679136/format-double-in-engineering-notation-with-fixed-number-of-fractional-digits), i had to put in a leading "%". Now i tried your suggestion without "%" (and it worked o c). Maybe the wiki could give some more examples and/or be more detailed regarding this point ...
anyway, thx again for the hint.

**Reply by dominiklenne** [reply]
Can i add this format string (or any other for that matter) to the drop down list for core text / format?

**Reply by macmarrum** [reply]
To quote the wiki 
> The predefined formats are defined in the file /formats.xml.

You can change `formats.xml` in your user_dir `Toos->Open user directory`, adding the ones you like. After restarting Freeplane, you'll see your formats in the Format drop-down.

**Reply by dominiklenne** [reply]
thx could have looked up this as well...
